src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -0,0 +1,416 @@
+            tags: [],
+            height: scaledRegionData.height,
+            width: scaledRegionData.width,
+            origin: {x: scaledRegionData.x, y: scaledRegionData.y},
+            points: scaledRegionData.points,
+        };
+
+        const currentAssetMetadata = this.props.selectedAsset;
+        currentAssetMetadata.regions.push(newRegion);
+        currentAssetMetadata.selectedRegions = [newRegion];
+        if (currentAssetMetadata.regions.length) {
+            currentAssetMetadata.asset.state = AssetState.Tagged;
+        }
+
+        this.props.onAssetMetadataChanged(currentAssetMetadata);
+    }
+
+    /**
+     * @name onRegionMove
+     * @description Method called when moving a region already in the editor
+     * @param {string} id the id of the region that was moved
+     * @param {RegionData} regionData the RegionData of moved region
+     * @returns {void}
+     */
+    public onRegionMove = (id: string, regionData: RegionData) => {
+        const ct = CanvasTools;
+        const currentAssetMetadata = this.props.selectedAsset;
+        const movedRegionIndex = currentAssetMetadata.regions.findIndex((region) => region.id === id);
+        const movedRegion = currentAssetMetadata.regions[movedRegionIndex];
+        // @ts-ignore   in here until CanvasTools types get updated
+        const scaledRegionData = this.scaleRegionToSourceSize(regionData);
+        if (movedRegion) {
+            movedRegion.points = scaledRegionData.points;
+        }
+        currentAssetMetadata.regions[movedRegionIndex] = movedRegion;
+        currentAssetMetadata.selectedRegions = [movedRegion];
+        this.props.onAssetMetadataChanged(currentAssetMetadata);
+    }
+
+    /**
+     * @name onRegionDelete
+     * @description Method called when deleting a region from the editor
+     * @param {string} id the id of the deleted region
+     * @returns {void}
+     */
+    public onRegionDelete = (id: string) => {
+        this.deleteRegionById(id);
+        const currentAssetMetadata = this.props.selectedAsset;
+        const deletedRegionIndex = this.props.selectedAsset.regions.findIndex((region) => region.id === id);
+        currentAssetMetadata.regions.splice(deletedRegionIndex, 1);
+        if (!currentAssetMetadata.regions.length) {
+            currentAssetMetadata.asset.state = AssetState.Visited;
+        }
+        currentAssetMetadata.selectedRegions = [];
+        this.props.onAssetMetadataChanged(currentAssetMetadata);
+    }
+
+    /**
+     * @name onRegionSelected
+     * @description Method called when deleting a region from the editor
+     * @param {string} id the id of the deleted region
+     * @param {boolean} multiselection boolean whether multiselect is active
+     * @returns {void}
+     */
+    public onRegionSelected = (id: string, multiselect: boolean) => {
+        const currentAssetMetadata = this.props.selectedAsset;
+        if (multiselect) {
+            currentAssetMetadata.selectedRegions.push(
```

Comment:
Recommend you move `selectedRegions` to be internal state to the canvas component vs messing with the assetMetadata.